--- x52pro-0.1.2/Makefile
+++ x52pro-0.1.2/Makefile
@@ -1,25 +1,31 @@
-CFLAGS=-Wall -g -I.
-LDFLAGS=-g
+CFLAGS+=-Wall -I.
 
 X52LIB=libx52pro.so.0.1.1
 
-all: $(X52LIB) x52output x52output.1.gz
+LBITS := $(shell getconf LONG_BIT)
+ifeq ($(LBITS),64)
+	LIBDIR=lib64
+else
+	LIBDIR=lib
+endif
+
+all: $(X52LIB) x52output
 
 $(X52LIB): x52pro.c
-	${CC} $< ${CFLAGS} -shared -Wl,-soname,libx52pro.so.0 -fPIC -D_REENTRANT -L. -lusb -o $@
+	${CC} $< ${CFLAGS} ${LDFLAGS} -shared -Wl,-soname,libx52pro.so.0 -fPIC -D_REENTRANT -L. -lusb -o $@
 
 clean:
 	-rm *.so* *.o x52output x52output.1.gz
 
 install:
 	install -D -m 644 x52pro.h $(DESTDIR)/usr/include/x52pro.h
-	install -D -m 644 $(X52LIB) $(DESTDIR)/usr/lib/$(X52LIB)
-	ln -sf $(X52LIB) $(DESTDIR)/usr/lib/libx52pro.so.0
-	ln -sf libx52pro.so.0 $(DESTDIR)/usr/lib/libx52pro.so
-	install -D -m 644 x52output.c $(DESTDIR)/usr/share/doc/libx52pro0/examples/x52output.c
+	install -D -m 644 $(X52LIB) $(DESTDIR)/usr/$(LIBDIR)/$(X52LIB)
+	ln -sf $(X52LIB) $(DESTDIR)/usr/$(LIBDIR)/libx52pro.so.0
+	ln -sf libx52pro.so.0 $(DESTDIR)/usr/$(LIBDIR)/libx52pro.so
+	install -D -m 644 x52output.c $(DESTDIR)/$(DOCDIR)/examples/x52output.c
 	install -D x52output $(DESTDIR)/usr/bin/x52output
-	install -D -m 644 99-x52pro.rules $(DESTDIR)/etc/udev/rules.d/99-x52pro.rules
-	install -D -m 644 x52output.1.gz $(DESTDIR)/usr/share/man/man1/x52output.1.gz
+	install -D -m 644 99-x52pro.rules $(DESTDIR)/lib/udev/rules.d/99-x52pro.rules
+	install -D -m 644 x52output.1 $(DESTDIR)/usr/share/man/man1/x52output.1
 
 x52output: x52output.o $(X52LIB) -lusb
 x52output.1.gz: x52output.1
