--- openssl-1.1.0-pre4/Makefile.in
+++ openssl-1.1.0-pre4/Makefile.in
@@ -379,7 +379,10 @@
 		libs="$$libs -l$$i"; \
 	done
 
-build-shared: do_$(SHLIB_TARGET) link-shared
+# The link target in Makefile.shared will create the symlink for us, so no need
+# to call link-shared directly.  Doing so will cause races with two processes
+# trying to symlink the lib.
+build-shared: do_$(SHLIB_TARGET)
 
 do_$(SHLIB_TARGET):
 	@ set -e; libs='-L. $(SHLIBDEPS)'; for i in $(SHLIBDIRS); do \
