--- libblockdev-2.7-1/src/lib/Makefile.am
+++ libblockdev-2.7-1/src/lib/Makefile.am
@@ -36,9 +36,10 @@
 BlockDev_2_0_gir_INCLUDES = GObject-2.0
 BlockDev_2_0_gir_PACKAGES = devmapper libudev libcryptsetup libparted blkid bytesize
 BlockDev_2_0_gir_CFLAGS = -I${builddir}/../../include/
-BlockDev_2_0_gir_LDFLAGS = -lm -ldmraid -L${builddir}/../utils/ -lbd_utils
+BlockDev_2_0_gir_LDFLAGS = -lm -L${builddir}/../utils/ -lbd_utils
 BlockDev_2_0_gir_SCANNERFLAGS = --warn-error --warn-all --identifier-prefix=BD --symbol-prefix=bd
 BlockDev_2_0_gir_EXPORT_PACKAGES = blockdev
+BlockDev_2_0_gir_LDLIBS =  -lm -ldmraid
 
 INTROSPECTION_GIRS = BlockDev-2.0.gir
 
