--- boost_1_69_0/boostcpp.jam
+++ boost_1_69_0/boostcpp.jam
@@ -123,7 +123,7 @@
 python-id = [ option.get "python-buildid" ] ;
 if $(python-id)
 {
-    PYTHON_ID = [ regex.replace $(python-id) "[*\\/:.\"\']" _ ] ;
+    PYTHON_ID = [ regex.replace $(python-id) "[*\\/:\"\']" _ ] ;
 }
 
 
--- boost_1_69_0/libs/python/Jamfile
+++ boost_1_69_0/libs/python/Jamfile
@@ -36,9 +36,10 @@
 # For example, Boost.Python built for Python 2.7 uses the suffix "27"
 rule version-suffix ( version )
 {
-    local major-minor = [ split-version $(version) ] ;
-    local suffix = $(major-minor:J="") ;
-    return $(suffix) ;
+    # local major-minor = [ split-version $(version) ] ;
+    # local suffix = $(major-minor:J="") ;
+    # return $(suffix) ;
+    return "" ;
 }
 
 
@@ -46,7 +47,7 @@
 python-id = [ option.get "python-buildid" ] ;
 if $(python-id)
 {
-    PYTHON_ID = [ regex.replace $(python-id) "[*\\/:.\"\']" _ ] ;
+    PYTHON_ID = [ regex.replace $(python-id) "[*\\/:\"\']" _ ] ;
 }
 
 rule python-tag ( name : type ? : property-set )
