From e411fcce53ae0fc2b1615da94a93f1985423f683 Mon Sep 17 00:00:00 2001
From: Lars Wendler <polynomial-c@gentoo.org>
Date: Mon, 18 May 2020 14:53:00 +0200
Subject: [PATCH] build: Don't install compressed man pages

---
 configure    |  4 ++--
 man/Makefile | 12 ++++++++++++
 2 files changed, 14 insertions(+), 2 deletions(-)

diff --git a/configure b/configure
index 2cdcd1b..e0d6790 100755
--- a/configure
+++ b/configure
@@ -367,7 +367,7 @@ LPNG_LIB  = $lpng_ldflags
 	
 	echo "
 all: cli gui
-cli: lib console plugins man" >> Makefile
+cli: lib console plugins" >> Makefile
 
 echo "
 lib:
@@ -408,7 +408,7 @@ cli_install:" >> Makefile
 	echo "	\$(MAKE) -C lib install
 	\$(MAKE) -C console install
 	\$(MAKE) -C plugins install
-	\$(MAKE) -C man install" >> Makefile
+	\$(MAKE) -C man install_unzipped" >> Makefile
 
 	echo "
 gui_install:" >> Makefile
diff --git a/man/Makefile b/man/Makefile
index 020ca18..b01fd51 100644
--- a/man/Makefile
+++ b/man/Makefile
@@ -44,6 +44,18 @@ install: f1tattoo.1.gz pxfw.8.gz cdvdcontrol.1.gz qscan.1.gz qscand.1.gz qpxtool
 	install -m 0644 qpxtool.1.gz $(MAN1DIR)
 	install -m 0644 readdvd.1.gz $(MAN1DIR)
 
+install_unzipped:
+	mkdir -p $(MAN1DIR)
+	mkdir -p $(MAN8DIR)
+	install -m 0644 f1tattoo.1 $(MAN1DIR)
+	install -m 0644 pxfw.8 $(MAN8DIR)
+	install -m 0644 cdvdcontrol.1 $(MAN1DIR)
+	install -m 0644 qscan.1 $(MAN1DIR)
+	install -m 0644 qscand.1 $(MAN1DIR)
+	install -m 0644 qpxtool.1 $(MAN1DIR)
+	install -m 0644 readdvd.1 $(MAN1DIR)
+
+
 uninstall:
 	rm -f $(MAN1DIR)/f1tattoo.1.gz
 	rm -f $(MAN8DIR)/pxfw.8.gz
-- 
2.26.2

