--- udev-init-scripts-19/helpers/write_root_link_rule
+++ udev-init-scripts-19/helpers/write_root_link_rule
@@ -0,0 +1,29 @@
+#!/bin/sh
+#
+# This script should run before doing "udevadm trigger" at boot.
+# It will create a rule matching the device directory / is on, and
+# creating /dev/root symlink pointing on its device node.
+#
+# This is especially useful for hal looking at /proc/mounts containing
+# a line listing /dev/root as device:
+#   /dev/root / reiserfs rw 0 0
+# 
+# This program is free software; you can redistribute it and/or modify it
+# under the terms of the GNU General Public License as published by the
+# Free Software Foundation version 2 of the License.
+#
+# (c) 2007-2008 Matthias Schwarzott <zzam@gentoo.org>
+
+eval $(udevadm info --export --export-prefix="ROOT_" --device-id-of-file=/)
+
+[ $? = 0 ] || exit 0
+[ "$ROOT_MAJOR" = 0 ] && exit 0
+
+[ -e /run ] && DIR=/run/udev/rules.d || DIR=/dev/.udev/rules.d
+[ -d "$DIR" ] || mkdir -p "$DIR"
+RULES=$DIR/10-root-link.rules
+
+echo "# Created by /lib/udev/write_root_link_rule" > "${RULES}"
+echo "# This rule should create /dev/root as link to real root device." >> "${RULES}"
+echo "SUBSYSTEM==\"block\", ENV{MAJOR}==\"$ROOT_MAJOR\", ENV{MINOR}==\"$ROOT_MINOR\", SYMLINK+=\"root\"" >> "${RULES}"
+
--- udev-init-scripts-19/init.d/udev
+++ udev-init-scripts-19/init.d/udev
@@ -72,6 +72,8 @@
 
 start_pre()
 {
+	%LIBUDEVDIR%/write_root_link_rule
+
 	check_kernel || return 1
 	if [ -e /proc/sys/kernel/hotplug ]; then
 		echo "" >/proc/sys/kernel/hotplug
--- udev-init-scripts-19/Makefile
+++ udev-init-scripts-19/Makefile
@@ -8,7 +8,10 @@
 CONFD ?= $(SYSCONFDIR)/conf.d
 INITD ?= $(SYSCONFDIR)/init.d
 
+SED ?= sed
+
 HELPERS = \
+	helpers/write_root_link_rule \
 	helpers/net.sh
 
 all:
@@ -22,6 +25,7 @@
 	install -m 0644 conf.d/* $(DESTDIR)$(CONFD)
 	install -d $(DESTDIR)$(INITD)
 	install -m 0755 init.d/* $(DESTDIR)$(INITD)
+	$(SED) "s|%LIBUDEVDIR%|$(LIBUDEV)|" -i $(DESTDIR)$(INITD)/udev
 
 check-git-repository:
 	git diff --quiet || { echo 'STOP, you have uncommitted changes in the working directory' ; false ; }
