--- arpalert-1.0.3/Makefile.in.orig	2006-09-01 23:53:40.000000000 +0200
+++ arpalert-1.0.3/Makefile.in	2006-09-06 15:00:47.686427360 +0200
@@ -23,7 +23,7 @@
 src_dir = .
 
 CC = @CC@
-CFLAGS = -Wall -O2  \
+MY_CFLAGS = -Wall $(CFLAGS)  \
          -DCONFIG_FILE=\"$(config_dir)/arpalert.conf\" \
          -DPID_FILE=\"$(lock_dir)/arpalert.pid\"
 LIBS = @LIBS@
@@ -40,39 +40,39 @@
 	$(src_dir)/capture.o
 
 arpalert: $(OBJECT_LIST)
-	$(CC) $(CFLAGS) -g -O2 -o arpalert $(OBJECT_LIST) $(LIBS)
+	$(CC) $(MY_CFLAGS) -g -o arpalert $(OBJECT_LIST) $(LIBS)
 
 alerte.o: alerte.c config.h alerte.h log.h loadconfig.h errmsg.h serveur.h
-	$(CC) $(CFLAGS) -c -o alerte.o $(src_dir)/alerte.c
+	$(CC) $(MY_CFLAGS) -c -o alerte.o $(src_dir)/alerte.c
 
 arpalert.o: arpalert.c config.h loadconfig.h log.h data.h maclist.h capture.h serveur.h \
             alerte.h sens.h sens_timeouts.h
-	$(CC) $(CFLAGS) -c -o arpalert.o $(src_dir)/arpalert.c
+	$(CC) $(MY_CFLAGS) -c -o arpalert.o $(src_dir)/arpalert.c
 
 capture.o: capture.c config.h capture.h sens.h log.h loadconfig.h data.h \
            alerte.h sens_timeouts.h
-	$(CC) $(CFLAGS) -c -o capture.o $(src_dir)/capture.c
+	$(CC) $(MY_CFLAGS) -c -o capture.o $(src_dir)/capture.c
 
 data.o: data.c config.h data.h log.h loadconfig.h
-	$(CC) $(CFLAGS) -c -o data.o $(src_dir)/data.c
+	$(CC) $(MY_CFLAGS) -c -o data.o $(src_dir)/data.c
 
 loadconfig.o: loadconfig.c loadconfig.h config.h log.h
-	$(CC) $(CFLAGS) -c -o loadconfig.o $(src_dir)/loadconfig.c
+	$(CC) $(MY_CFLAGS) -c -o loadconfig.o $(src_dir)/loadconfig.c
 
 log.o: log.c log.h config.h loadconfig.h
-	$(CC) $(CFLAGS) -c -o log.o $(src_dir)/log.c $(DEFS)
+	$(CC) $(MY_CFLAGS) -c -o log.o $(src_dir)/log.c $(DEFS)
 
 maclist.o: maclist.c maclist.h config.h data.h loadconfig.h log.h
-	$(CC) $(CFLAGS) -c -o maclist.o $(src_dir)/maclist.c
+	$(CC) $(MY_CFLAGS) -c -o maclist.o $(src_dir)/maclist.c
 
 sens.o: sens.c sens.h config.h loadconfig.h data.h log.h
-	$(CC) $(CFLAGS) -c -o sens.o $(src_dir)/sens.c
+	$(CC) $(MY_CFLAGS) -c -o sens.o $(src_dir)/sens.c
 
 sens_timeouts.o: sens_timeouts.c sens_timeouts.h config.h data.h log.h loadconfig.h
-	$(CC) $(CFLAGS) -c -o sens_timeouts.o $(src_dir)/sens_timeouts.c
+	$(CC) $(MY_CFLAGS) -c -o sens_timeouts.o $(src_dir)/sens_timeouts.c
 
 serveur.o: serveur.c serveur.h config.h log.h loadconfig.h
-	$(CC) $(CFLAGS) -c -o serveur.o $(src_dir)/serveur.c
+	$(CC) $(MY_CFLAGS) -c -o serveur.o $(src_dir)/serveur.c
 
 mrproper: clean
 	rm -rf arpalert-sources.pdf autom4te.cache configure Makefile \
