--- lsiutil/lsiutil.c
+++ lsiutil/lsiutil.c
@@ -1581,7 +1581,7 @@
 
     if (argc > 1)
     {
-        while ((arg = getopt(argc, argv, "?a:bc:def:ghij:kl:m:np:qrst:uv:wxyz018")) != EOF)
+        while ((arg = getopt(argc, argv, "?a:bc:def:ghij:kl:m:np:qrst:uv:wxyzV018")) != EOF)
         {
             switch (arg)
             {
@@ -1614,6 +1614,10 @@
 "      -b                Show board manufacturing information.\n"
 "      -m freq[,time]    Monitor port performance, updating the display\n"
 "                        every 'freq' seconds, for 'time' seconds.\n\n");
+				printf(
+"\n"
+"      -h                Shows this help text\n"
+"      -V                Prints the program name and version\n");
                 printf(
 "Examples:\n"
 "1. to display the port settings and targets for port 1:\n"
@@ -1655,6 +1659,10 @@
 );
                 return 0;
 
+			case 'V':
+				/* already done before options parsing */
+				return 0;
+
             case 'a':
                 args = optarg;
                 break;
