--- libnitrokey-3.0/unittest/catch_main.cpp
+++ libnitrokey-3.0/unittest/catch_main.cpp
@@ -1,2 +1,2 @@
 #define CATCH_CONFIG_MAIN  // This tells Catch to provide a main()
-#include "catch.hpp"
\ No newline at end of file
+#include <catch/catch.hpp>
--- libnitrokey-3.0/unittest/test_offline.cc
+++ libnitrokey-3.0/unittest/test_offline.cc
@@ -1,4 +1,4 @@
-#include "catch.hpp"
+#include <catch/catch.hpp>
 #include <NitrokeyManager.h>
 #include <memory>
 #include "../NK_C_API.h"
