--- lxsession-0.4.9.2/configure.ac
+++ lxsession-0.4.9.2/configure.ac
@@ -81,11 +81,6 @@
 AC_SUBST(DBUSGLIB_CFLAGS)
 AC_SUBST(DBUSGLIB_LIBS)
 
-polkit_modules="polkit-agent-1"
-PKG_CHECK_MODULES(POLKIT, [$polkit_modules])
-AC_SUBST(POLKIT_CFLAGS)
-AC_SUBST(POLKIT_LIBS)
-
 PKG_CHECK_MODULES(DBUS, [dbus-1])
 AC_SUBST(DBUS_CFLAGS)
 AC_SUBST(DBUS_LIBS)
--- lxsession-0.4.9.2/Makefile.am
+++ lxsession-0.4.9.2/Makefile.am
@@ -5,7 +5,6 @@
 SUBDIRS= data \
          lxclipboard \
          lxlock \
-         lxpolkit \
          lxsession \
          lxsession-default \
          lxsession-default-apps \
