--- e2fsprogs-1.43/lib/blkid/devname.c
+++ e2fsprogs-1.43/lib/blkid/devname.c
@@ -36,6 +36,9 @@
 #if HAVE_SYS_MKDEV_H
 #include <sys/mkdev.h>
 #endif
+#ifdef HAVE_SYS_SYSMACROS_H
+#include <sys/sysmacros.h>
+#endif
 #include <time.h>
 
 #include "blkidP.h"
--- e2fsprogs-1.43/lib/blkid/devno.c
+++ e2fsprogs-1.43/lib/blkid/devno.c
@@ -31,6 +31,9 @@
 #if HAVE_SYS_MKDEV_H
 #include <sys/mkdev.h>
 #endif
+#ifdef HAVE_SYS_SYSMACROS_H
+#include <sys/sysmacros.h>
+#endif
 
 #include "blkidP.h"
 
--- e2fsprogs-1.43/lib/ext2fs/finddev.c
+++ e2fsprogs-1.43/lib/ext2fs/finddev.c
@@ -31,6 +31,9 @@
 #if HAVE_SYS_MKDEV_H
 #include <sys/mkdev.h>
 #endif
+#ifdef HAVE_SYS_SYSMACROS_H
+#include <sys/sysmacros.h>
+#endif
 
 #include "ext2_fs.h"
 #include "ext2fs.h"
--- e2fsprogs-1.43/lib/ext2fs/ismounted.c
+++ e2fsprogs-1.43/lib/ext2fs/ismounted.c
@@ -49,6 +49,9 @@
 #if HAVE_SYS_TYPES_H
 #include <sys/types.h>
 #endif
+#ifdef HAVE_SYS_SYSMACROS_H
+#include <sys/sysmacros.h>
+#endif
 
 #include "ext2_fs.h"
 #include "ext2fs.h"
--- e2fsprogs-1.43/misc/mk_hugefiles.c
+++ e2fsprogs-1.43/misc/mk_hugefiles.c
@@ -35,6 +35,9 @@
 #include <sys/ioctl.h>
 #include <sys/types.h>
 #include <sys/stat.h>
+#ifdef HAVE_SYS_SYSMACROS_H
+#include <sys/sysmacros.h>
+#endif
 #include <libgen.h>
 #include <limits.h>
 #include <blkid/blkid.h>
