--- mgba-0.4.1/CMakeLists.txt
+++ mgba-0.4.1/CMakeLists.txt
@@ -450,30 +450,7 @@
 endif()
 
 if (USE_LZMA)
-	include_directories(AFTER ${CMAKE_SOURCE_DIR}/third-party/lzma)
 	add_definitions(-D_7ZIP_PPMD_SUPPPORT)
-	list(APPEND VFS_SRC ${CMAKE_SOURCE_DIR}/src/util/vfs/vfs-lzma.c)
-	set(LZMA_SRC
-		${CMAKE_SOURCE_DIR}/src/third-party/lzma/7zAlloc.c
-		${CMAKE_SOURCE_DIR}/src/third-party/lzma/7zArcIn.c
-		${CMAKE_SOURCE_DIR}/src/third-party/lzma/7zBuf.c
-		${CMAKE_SOURCE_DIR}/src/third-party/lzma/7zBuf2.c
-		${CMAKE_SOURCE_DIR}/src/third-party/lzma/7zCrc.c
-		${CMAKE_SOURCE_DIR}/src/third-party/lzma/7zCrcOpt.c
-		${CMAKE_SOURCE_DIR}/src/third-party/lzma/7zDec.c
-		${CMAKE_SOURCE_DIR}/src/third-party/lzma/CpuArch.c
-		${CMAKE_SOURCE_DIR}/src/third-party/lzma/Delta.c
-		${CMAKE_SOURCE_DIR}/src/third-party/lzma/LzmaDec.c
-		${CMAKE_SOURCE_DIR}/src/third-party/lzma/Lzma2Dec.c
-		${CMAKE_SOURCE_DIR}/src/third-party/lzma/Bra.c
-		${CMAKE_SOURCE_DIR}/src/third-party/lzma/Bra86.c
-		${CMAKE_SOURCE_DIR}/src/third-party/lzma/BraIA64.c
-		${CMAKE_SOURCE_DIR}/src/third-party/lzma/Bcj2.c
-		${CMAKE_SOURCE_DIR}/src/third-party/lzma/Ppmd7.c
-		${CMAKE_SOURCE_DIR}/src/third-party/lzma/Ppmd7Dec.c
-		${CMAKE_SOURCE_DIR}/src/third-party/lzma/7zFile.c
-		${CMAKE_SOURCE_DIR}/src/third-party/lzma/7zStream.c)
-	list(APPEND VFS_SRC ${LZMA_SRC})
 	list(APPEND FEATURES LZMA)
 endif()
 
