--- mc-4.8.16/misc/syntax/ebuild.syntax
+++ mc-4.8.16/misc/syntax/ebuild.syntax
@@ -75,6 +75,8 @@
     keyword whole dosed yellow
     keyword whole dosym yellow
     keyword whole eaclocal yellow
+    keyword whole eapply yellow
+    keyword whole eapply_user yellow
     keyword whole eautoconf yellow
     keyword whole eautoreconf yellow
     keyword whole eautomake yellow
@@ -84,6 +86,7 @@
     keyword whole eerror yellow
     keyword whole einfo yellow
     keyword whole einstall yellow
+    keyword whole einstalldocs yellow
     keyword whole elibtoolize yellow
     keyword whole elog yellow
     keyword whole emake yellow
