--- nitrokey-app-1.2.1/src/core/ThreadWorker.cpp
+++ nitrokey-app-1.2.1/src/core/ThreadWorker.cpp
@@ -2,7 +2,7 @@
 
 #include <QMutexLocker>
 #include <QDebug>
-#include "libnitrokey/include/DeviceCommunicationExceptions.h"
+#include <libnitrokey/DeviceCommunicationExceptions.h>
 
 
 ThreadWorkerNS::Worker::Worker(QObject *parent, const std::function<Data()> &datafunc) :
--- nitrokey-app-1.2.1/src/GUI/Authentication.cpp
+++ nitrokey-app-1.2.1/src/GUI/Authentication.cpp
@@ -3,7 +3,7 @@
 #include "Authentication.h"
 #include "pindialog.h"
 #include "libada.h"
-#include "libnitrokey/include/NitrokeyManager.h"
+#include <libnitrokey/NitrokeyManager.h>
 #include "nitrokey-applet.h"
 
 using nm = nitrokey::NitrokeyManager;
--- nitrokey-app-1.2.1/src/GUI/StorageActions.cpp
+++ nitrokey-app-1.2.1/src/GUI/StorageActions.cpp
@@ -2,7 +2,7 @@
 #include "StorageActions.h"
 #include "src/ui/pindialog.h"
 #include "src/utils/bool_values.h"
-#include "libnitrokey/include/NitrokeyManager.h"
+#include <libnitrokey/NitrokeyManager.h>
 
 #ifdef Q_OS_LINUX
 #include "systemutils.h"
@@ -19,7 +19,7 @@
 #define LOCAL_PASSWORD_SIZE 40
 #include <memory>
 #include <src/core/ThreadWorker.h>
-#include <libnitrokey/include/stick20_commands.h>
+#include <libnitrokey/stick20_commands.h>
 
 
 void unmountEncryptedVolumes() {
--- nitrokey-app-1.2.1/src/GUI/Tray.cpp
+++ nitrokey-app-1.2.1/src/GUI/Tray.cpp
@@ -9,7 +9,7 @@
 #include "libada.h"
 #include "bool_values.h"
 #include "nitrokey-applet.h"
-#include <libnitrokey/include/NitrokeyManager.h>
+#include <libnitrokey/NitrokeyManager.h>
 #include <QMenu>
 
 Tray::Tray(QObject *_parent, bool _debug_mode, bool _extended_config,
--- nitrokey-app-1.2.1/src/libada.cpp
+++ nitrokey-app-1.2.1/src/libada.cpp
@@ -1,4 +1,4 @@
-#include <libnitrokey/include/NitrokeyManager.h>
+#include <libnitrokey/NitrokeyManager.h>
 #include "libada.h"
 
 
@@ -77,7 +77,7 @@
 }
 
 #include <QtCore/QMutex>
-#include "libnitrokey/include/CommandFailedException.h"
+#include <libnitrokey/CommandFailedException.h>
 #include "hotpslot.h"
 
 void libada::on_FactoryReset(){
@@ -184,7 +184,7 @@
 }
 
 #include <QDebug>
-#include "libnitrokey/include/DeviceCommunicationExceptions.h"
+#include <libnitrokey/DeviceCommunicationExceptions.h>
 bool libada::isPasswordSafeUnlocked() {
   try{
     nm::instance()->get_password_safe_slot_status();
--- nitrokey-app-1.2.1/src/libada.h
+++ nitrokey-app-1.2.1/src/libada.h
@@ -36,7 +36,7 @@
 #include <QObject>
 #include <QCache>
 
-#include "libnitrokey/include/stick10_commands.h"
+#include <libnitrokey/stick10_commands.h>
 
 #include <functional>
 #include <QMutex>
--- nitrokey-app-1.2.1/src/main.cpp
+++ nitrokey-app-1.2.1/src/main.cpp
@@ -26,7 +26,7 @@
 #include <QSettings>
 #include <QFileInfo>
 #include <QDebug>
-#include <libnitrokey/include/log.h>
+#include <libnitrokey/log.h>
 //#include <QSharedMemory>
 #include "src/version.h"
 #include "src/utils/bool_values.h"
--- nitrokey-app-1.2.1/src/ui/aboutdialog.cpp
+++ nitrokey-app-1.2.1/src/ui/aboutdialog.cpp
@@ -23,8 +23,8 @@
 #include "ui_aboutdialog.h"
 
 #include "libada.h"
-#include "libnitrokey/include/NitrokeyManager.h"
-#include "libnitrokey/include/DeviceCommunicationExceptions.h"
+#include <libnitrokey/NitrokeyManager.h>
+#include <libnitrokey/DeviceCommunicationExceptions.h>
 using nm = nitrokey::NitrokeyManager;
 
 using namespace AboutDialogUI;
@@ -93,7 +93,7 @@
 #include <QTextStream>
 #include <QWindow>
 #include <QTextEdit>
-#include <libnitrokey/include/stick20_commands.h>
+#include <libnitrokey/stick20_commands.h>
 #include "licensedialog.h"
 
 
--- nitrokey-app-1.2.1/src/ui/mainwindow.cpp
+++ nitrokey-app-1.2.1/src/ui/mainwindow.cpp
@@ -17,7 +17,7 @@
  * You should have received a copy of the GNU General Public License
  * along with Nitrokey. If not, see <http://www.gnu.org/licenses/>.
  */
-#include "libnitrokey/include/NitrokeyManager.h"
+#include <libnitrokey/NitrokeyManager.h>
 
 #include "mainwindow.h"
 #include "aboutdialog.h"
--- nitrokey-app-1.2.1/src/ui/stick20changepassworddialog.cpp
+++ nitrokey-app-1.2.1/src/ui/stick20changepassworddialog.cpp
@@ -22,7 +22,7 @@
 #include "stick20responsedialog.h"
 #include "ui_stick20changepassworddialog.h"
 #include "libada.h"
-#include "libnitrokey/include/NitrokeyManager.h"
+#include <libnitrokey/NitrokeyManager.h>
 #include "src/core/ThreadWorker.h"
 using nm = nitrokey::NitrokeyManager;
 
--- nitrokey-app-1.2.1/src/ui/stick20hiddenvolumedialog.cpp
+++ nitrokey-app-1.2.1/src/ui/stick20hiddenvolumedialog.cpp
@@ -18,7 +18,7 @@
  */
 
 #include <src/core/ThreadWorker.h>
-#include <libnitrokey/include/NitrokeyManager.h>
+#include <libnitrokey/NitrokeyManager.h>
 #include "stick20hiddenvolumedialog.h"
 #include "math.h"
 #include "mcvs-wrapper.h"
--- nitrokey-app-1.2.1/src/ui/stick20responsedialog.cpp
+++ nitrokey-app-1.2.1/src/ui/stick20responsedialog.cpp
@@ -106,7 +106,7 @@
 //  delete pollStick20Timer;
 }
 
-#include "libnitrokey/include/NitrokeyManager.h"
+#include <libnitrokey/NitrokeyManager.h>
 
 void Stick20ResponseDialog::checkStick20StatusDialog(void) {
 //  QString OutputText;
