--- hplip-3.17.11/configure.in
+++ hplip-3.17.11/configure.in
@@ -225,6 +225,7 @@
 else
    AC_MSG_RESULT(no)
 fi
+AM_CONDITIONAL(PP_BUILD, test x$pp_build = xyes)
 AM_CONDITIONAL(NETWORK_BUILD, test x$network_build = xyes)
 
 AC_MSG_CHECKING([for parallel port build])
--- hplip-3.17.11/Makefile.am
+++ hplip-3.17.11/Makefile.am
@@ -117,6 +117,10 @@
 endif
 endif #HPLIP_CLASS_DRIVER
 
+if PP_BUILD
+libhpmud_la_LDFLAGS = -lrt
+endif
+
 # ip library
 lib_LTLIBRARIES += libhpip.la
 libhpip_la_LDFLAGS = -version-info 0:1:0
