From 9b0dc913f7f29837512803bfbbd564dc33212000 Mon Sep 17 00:00:00 2001
From: Lars Wendler <polynomial-c@gentoo.org>
Date: Mon, 18 Jan 2021 15:47:34 +0100
Subject: [PATCH] udev_helper/net.sh: Ignore wireguard interfaces

Bug: https://bugs.gentoo.org/758938
Signed-off-by: Lars Wendler <polynomial-c@gentoo.org>
---
 udev_helper/net.sh | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/udev_helper/net.sh b/udev_helper/net.sh
index f1c4f20..0c114a5 100644
--- a/udev_helper/net.sh
+++ b/udev_helper/net.sh
@@ -17,7 +17,7 @@ fi
 
 # ignore interfaces that are registered after being "up" (?)
 case ${IFACE} in
-	ppp*|ippp*|isdn*|plip*|lo*|irda*|dummy*|ipsec*|tun*|tap*|br*|macvtap*|ipvl*|vlan*|vxlan*|peth*|ifb*|veth*|gre*|vrf*)
+	ppp*|ippp*|isdn*|plip*|lo*|irda*|dummy*|ipsec*|tun*|tap*|br*|macvtap*|ipvl*|vlan*|vxlan*|peth*|ifb*|veth*|gre*|vrf*|wg*)
 		exit 0 ;;
 esac
 
-- 
2.30.0

