--- logrotate-3.12.1/config.c
+++ logrotate-3.12.1/config.c
@@ -1,6 +1,6 @@
 #include "queue.h"
 /* Alloca is defined in stdlib.h in NetBSD */
-#ifndef __NetBSD__
+#if !defined(__NetBSD__) && !defined(__FreeBSD__)
 #include <alloca.h>
 #endif
 #include <limits.h>
@@ -25,6 +25,10 @@
 #include <sys/mman.h>
 #include <libgen.h>
 
+#if !defined(PATH_MAX) && defined(__FreeBSD__)
+#include <sys/param.h>
+#endif
+
 #include "log.h"
 #include "logrotate.h"
 
--- logrotate-3.12.1/logrotate.c
+++ logrotate-3.12.1/logrotate.c
@@ -1,6 +1,6 @@
 #include "queue.h"
 /* alloca() is defined in stdlib.h in NetBSD */
-#ifndef __NetBSD__
+#if !defined(__NetBSD__) && !defined(__FreeBSD__)
 #include <alloca.h>
 #endif
 #include <limits.h>
@@ -27,6 +27,10 @@
 #include <limits.h>
 #endif
 
+#if !defined(PATH_MAX) && defined(__FreeBSD__)
+#include <sys/param.h>
+#endif
+
 #include "log.h"
 #include "logrotate.h"
 
