--- logrotate-3.15.0/config.c
+++ logrotate-3.15.0/config.c
@@ -417,7 +417,9 @@
     int i;
 
     /* Check if fname is '.' or '..'; if so, return false */
-    if (fname[0] == '.' && (!fname[1] || (fname[1] == '.' && !fname[2])))
+    /* Don't include 'hidden' files either; this breaks Gentoo
+       portage config file management http://bugs.gentoo.org/87683 */
+    if (fname[0] == '.')
         return 0;
 
     /* Check if fname is ending in a taboo-extension; if so, return false */
