https://github.com/marazmista/radeon-profile/issues/244

--- a/radeon-profile/gpu.cpp
+++ b/radeon-profile/gpu.cpp
@@ -69,7 +69,13 @@ void gpu::detectCards() {
 
         int pciIdx = ueventContents.indexOf(QRegExp("PCI_SLOT_NAME.+"));
         if (pciIdx != -1)
-            gsi.name = globalStuff::grabSystemInfo("lspci -s " + ueventContents[pciIdx].split('=')[1])[0].split(':')[2].trimmed();
+        {
+            auto input = globalStuff::grabSystemInfo("lspci -s " + ueventContents[pciIdx].split('=')[1]);
+            if(input.size() < 3)
+                gsi.name = "UNKNOWN";
+            else
+                gsi.name = input[0].split(':')[2].trimmed();
+        }
 
         gpuList.append(gsi);
 
