From 600734113ca56d233ef795f9f2e4d83ba7bbdd61 Mon Sep 17 00:00:00 2001
From: IR4T4 <gutzu@gmx.ch>
Date: Thu, 24 Jan 2019 10:49:29 +0100
Subject: [PATCH] server: fix build with FEATURE_IPV6 enabled - many thanks @
 Poly-C!

---
 src/server/sv_main.c | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/src/server/sv_main.c b/src/server/sv_main.c
index 3ca492663..4fa7ac30e 100644
--- a/src/server/sv_main.c
+++ b/src/server/sv_main.c
@@ -405,7 +405,7 @@ void SV_MasterHeartbeat(const char *msg)
 #ifdef FEATURE_IPV6
 		if (netenabled & NET_ENABLEV6 && adr[i][1].type != NA_BAD)
 		{
-			NET_OutOfBandPrint(NS_SERVER, adr[i][1], "heartbeat %s\n", message);
+			NET_OutOfBandPrint(NS_SERVER, adr[i][1], "heartbeat %s\n", msg);
 		}
 #endif
 	}
