--- gkrellm2-cpupower-0.1.5/cpufreqnextgovernor
+++ gkrellm2-cpupower-0.1.5/cpufreqnextgovernor
@@ -3,10 +3,10 @@
 cpu="$1"
 
 # determine actual governor
-actual_gov=`cpupower -c $cpu frequency-info -p | tail -n 1 | sed 's/.*\ //'`
+actual_gov=`cpupower -c $cpu frequency-info -p | sed -n '/governor/s@.*"\([[:alpha:]]\+\)".*@\1@p'`
 
 # determine available governors
-avail_gov=`cpupower -c $cpu frequency-info -g | tail -n 1`
+avail_gov=`cpupower -c $cpu frequency-info -g | sed -n '/available/s@.*: @@p'`
 
 avail_gov1=`echo $avail_gov | sed "s/.*$actual_gov//"`
 avail_gov2=`echo $avail_gov | sed "s/$actual_gov.*//"`
