--- sudosh2-1.0.5/src/Makefile.am
+++ sudosh2-1.0.5/src/Makefile.am
@@ -8,6 +8,6 @@
 
 install-sudosh.conf:
 	test -z "$(sysconfdir)" || $(mkdir_p) "$(DESTDIR)$(sysconfdir)"
-	test -f $(sysconfdir)/sudosh.conf || $(INSTALL) -o 0 -g 0 -m 0444 '$(srcdir)/sudosh.conf' '$(DESTDIR)$(sysconfdir)/sudosh.conf'
+	test -f $(sysconfdir)/sudosh.conf || $(INSTALL) -o 0 -g 0 -m 0444 $(srcdir)/sudosh.conf $(DESTDIR)$(sysconfdir)/sudosh.conf
 
 install: install-am install-sudosh.conf
