* Fix build system when USE="-threads"

--- xerces-c-3.2.2/configure.ac
+++ xerces-c-3.2.2/configure.ac
@@ -159,6 +159,7 @@
 AC_CXX_HAVE_LSTRING
 
 ACX_PTHREAD
+AM_CONDITIONAL([THREADS], [test "x${acx_pthread_ok}" = "xyes"])
 
 # Checks for library functions.
 #AC_FUNC_ERROR_AT_LINE
--- xerces-c-3.2.2/tests/Makefile.am
+++ xerces-c-3.2.2/tests/Makefile.am
@@ -21,7 +21,7 @@
 
 check_PROGRAMS =                                ${testprogs}
 
-LDADD =	                                        ${top_builddir}/src/libxerces-c.la
+LDADD =	                                        $(top_builddir)/src/libxerces-c.la $(PTHREAD_LIBS)
 # the  -I${top_builddir}/src is needed to pick the xercesc/util/Xerces_autoconf_config.hpp 
 # header file when doing out-of-tree builds
 AM_CPPFLAGS =                                   -I${top_builddir}/src -I${top_srcdir}/src
@@ -92,8 +92,10 @@
 #                                               src/ParserTest/ParserTest_Parser.cpp \
 #                                               src/ParserTest/ParserTest_Parser.hpp
 
+if THREADS
 testprogs +=                                    ThreadTest
 ThreadTest_SOURCES =                            src/ThreadTest/ThreadTest.cpp
+endif
 
 # Fails to compile under gcc 4 (ambiguous calls to NullPointerException)
 # dcargill says this is obsolete and we can delete it.
