From 68c4967e20c1cb2a4c2a345d3e122adaa5d1b3d3 Mon Sep 17 00:00:00 2001
From: =?UTF-8?q?Holger=20Hoffst=C3=A4tte?= <holger@applied-asynchrony.com>
Date: Thu, 28 Jun 2018 23:08:57 +0200
Subject: [PATCH] rc-status: prevent memory corruption by freeing uninitialized
 pointer

---
 src/rc/rc-status.c | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/src/rc/rc-status.c b/src/rc/rc-status.c
index 85e71dbb..cebdc5ed 100644
--- a/src/rc/rc-status.c
+++ b/src/rc/rc-status.c
@@ -87,7 +87,7 @@ static char *get_uptime(const char *service)
 	time_t diff_hours = (time_t) 0;
 	time_t diff_mins = (time_t) 0;
 	time_t diff_secs = (time_t) 0;
-	char *uptime;
+	char *uptime = NULL;
 
 	if (state & RC_SERVICE_STARTED) {
 		start_count = rc_service_value_get(service, "start_count");
