From 45130a237874aaa96bdf23ee107b0be0e0a2afab Mon Sep 17 00:00:00 2001
From: Alex Kontos <MrAlex94@users.noreply.github.com>
Date: Fri, 10 Jan 2020 17:58:12 +0000
Subject: [PATCH] Waterfox doesn't support BigInt, so remove references to it.

---
 js/xpconnect/src/XPCVariant.cpp | 6 +++---
 1 file changed, 3 insertions(+), 3 deletions(-)

diff --git a/js/xpconnect/src/XPCVariant.cpp b/js/xpconnect/src/XPCVariant.cpp
index 9683a4cce757..44794880181b 100644
--- a/js/xpconnect/src/XPCVariant.cpp
+++ b/js/xpconnect/src/XPCVariant.cpp
@@ -180,7 +180,7 @@ XPCArrayHomogenizer::GetTypeForArray(JSContext* cx, HandleObject array,
             type = tDbl;
         } else if (val.isBoolean()) {
             type = tBool;
-        } else if (val.isUndefined() || val.isSymbol() || val.isBigInt()) {
+        } else if (val.isUndefined() || val.isSymbol()) {
             state = tVar;
             break;
         } else if (val.isNull()) {
@@ -275,8 +275,8 @@ bool XPCVariant::InitializeData(JSContext* cx)
         mData.SetFromBool(val.toBoolean());
         return true;
     }
-    // We can't represent symbol or BigInt on C++ side, so pretend it is void.
-    if (val.isUndefined() || val.isSymbol() || val.isBigInt()) {
+    // We can't represent symbol on C++ side, so pretend it is void.
+    if (val.isUndefined() || val.isSymbol()) {
         mData.SetToVoid();
         return true;
     }
-- 
2.25.0

