From cdf98ceb2b792c5d93cedc9b8ef8be83fd749ec7 Mon Sep 17 00:00:00 2001
From: Lars Wendler <polynomial-c@gentoo.org>
Date: Fri, 26 Mar 2021 20:15:06 +0100
Subject: [PATCH] 2021.03 build fix

Signed-off-by: Lars Wendler <polynomial-c@gentoo.org>
---
 gfx/angle/checkout/src/common/mathutil.h | 1 -
 1 file changed, 1 deletion(-)

diff --git a/gfx/angle/checkout/src/common/mathutil.h b/gfx/angle/checkout/src/common/mathutil.h
index 1d73bbffff44..70149ec2d096 100644
--- a/gfx/angle/checkout/src/common/mathutil.h
+++ b/gfx/angle/checkout/src/common/mathutil.h
@@ -1386,7 +1386,6 @@ inline uint16_t RotR16(uint16_t x, int8_t r)
 constexpr unsigned int Log2(unsigned int bytes)
 {
     return bytes == 1 ? 0 : (1 + Log2(bytes / 2));
-}
 }  // namespace rx
 
 #endif  // COMMON_MATHUTIL_H_
-- 
2.31.0

