--- nikto-2.1.2/nikto.pl
+++ nikto-2.1.2/nikto.pl
@@ -52,7 +52,7 @@
 $NIKTO{'DIV'}     = "-" x 75;
 $NIKTO{'version'} = "2.1.2";
 $NIKTO{'name'}    = "Nikto";
-$NIKTO{'configfile'} = "/etc/nikto.conf";    ### Change this line if it's having trouble finding it
+$NIKTO{'configfile'} = "/etc/nikto/nikto.conf";    ### Change this line if it's having trouble finding it
 
 # put a signal trap so we can close down reports properly
 $SIG{'INT'} = \&safe_quit;
@@ -89,7 +89,8 @@
 require "$NIKTOCONFIG{'PLUGINDIR'}/nikto_core.plugin";
 nprint("T:$starttime: Starting", "d");
 require "$NIKTOCONFIG{'PLUGINDIR'}/nikto_single.plugin";
-require "$NIKTOCONFIG{'PLUGINDIR'}/LW2.pm";
+
+use LW2;
 
 my ($a, $b) = split(/\./, $LW2::VERSION);
 die("- You must use LW2 2.4 or later\n") if ($a != 2 || $b < 4);
