--- mozilla/netwerk/base/nsSocketTransportService2.cpp	2021-02-02 03:17:03.301412550 +0300
+++ mozilla/netwerk/base/nsSocketTransportService2.cpp-Maybe	2021-02-02 03:23:09.859417588 +0300
@@ -531,17 +531,17 @@
     // DoPollIteration() should service the network without blocking.
     bool pendingEvents = false;
     mRawThread->HasPendingEvents(&pendingEvents);
 
     if (mPollList[0].fd) {
         mPollList[0].out_flags = 0;
         pollList = mPollList;
         pollCount = mActiveCount + 1;
-        pollTimeout = pendingEvents ? PR_INTERVAL_NO_WAIT : PollTimeout(ts);
+        pollTimeout = pendingEvents ? PR_MillisecondsToInterval(123) : PollTimeout(ts);
     }
     else {
         // no pollable event, so busy wait...
         pollCount = mActiveCount;
         if (pollCount)
             pollList = &mPollList[1];
         else
             pollList = nullptr;
