--- check_multi-0.25/contrib/notify_service/nail/Makefile
+++ check_multi-0.25/contrib/notify_service/nail/Makefile
@@ -63,7 +63,7 @@
 ###########################################################################
 ###########################################################################
 
-FEATURES	= -DMAILRC='"$(MAILRC)"' -DMAILSPOOL='"$(MAILSPOOL)"' \
+NAIL_FEATURES	= -DMAILRC='"$(MAILRC)"' -DMAILSPOOL='"$(MAILSPOOL)"' \
 			-DSENDMAIL='"$(SENDMAIL)"' $(IPv6)
 
 OBJ = aux.o base64.o cache.o cmd1.o cmd2.o cmd3.o cmdtab.o collect.o \
@@ -77,13 +77,13 @@
 
 .SUFFIXES: .o .c .x
 .c.o:
-	$(CC) $(CFLAGS) $(CPPFLAGS) $(FEATURES) $(INCLUDES) $(WARN) -c $<
+	$(CC) $(CFLAGS) $(CPPFLAGS) $(NAIL_FEATURES) $(INCLUDES) $(WARN) -c $<
 
 .c.x:
-	$(CC) $(CFLAGS) $(CPPFLAGS) $(FEATURES) $(INCLUDES) $(WARN) -E $< >$@
+	$(CC) $(CFLAGS) $(CPPFLAGS) $(NAIL_FEATURES) $(INCLUDES) $(WARN) -E $< >$@
 
 .c:
-	$(CC) $(CFLAGS) $(CPPFLAGS) $(FEATURES) $(INCLUDES) $(WARN) \
+	$(CC) $(CFLAGS) $(CPPFLAGS) $(NAIL_FEATURES) $(INCLUDES) $(WARN) \
 		$(LDFLAGS) $< `grep '^[^#]' LIBS` $(LIBS) -o $@
 
 all: mailx
