From 657be04fea405cd721d7e1197980b790b0bdd41e Mon Sep 17 00:00:00 2001
From: orbea <orbea@riseup.net>
Date: Jan 28 2021 15:02:57 +0000
Subject: slbt_exec_install_entry(): Allow using both -release and -version-info.


Fixes https://dev.midipix.org/cross/slibtool/issue/26

---

diff --git a/src/logic/slbt_exec_install.c b/src/logic/slbt_exec_install.c
index b8d3ee9..eadf76a 100644
--- a/src/logic/slbt_exec_install.c
+++ b/src/logic/slbt_exec_install.c
@@ -337,7 +337,6 @@ static int slbt_exec_install_entry(
 	char		lasource[PATH_MAX - 8];
 	bool		fexe = false;
 	bool		fpe;
-	bool		frelease;
 	struct stat	st;
 
 	/* executable wrapper? */
@@ -407,7 +406,6 @@ static int slbt_exec_install_entry(
 
 	/* libfoo.a --> libfoo.so.release */
 	sprintf(dot,"%s.release",dctx->cctx->settings.dsosuffix);
-	frelease = stat(slnkname,&st) ? false : true;
 
 	/* libfoo.a --> libfoo.so */
 	strcpy(dot,dctx->cctx->settings.dsosuffix);
@@ -487,9 +485,6 @@ static int slbt_exec_install_entry(
 			false))
 		return SLBT_NESTED_ERROR(dctx);
 
-	if (frelease)
-		return 0;
-
 	/* libfoo.so.x --> libfoo.so.x.y.z */
 	strcpy(slnkname,target);
 

