https://code.google.com/p/clementine-player/issues/detail?id=3847#c27

diff --git a/src/library/libraryquery.cpp b/src/library/libraryquery.cpp
index 16e7ead..026f7b0 100644
--- a/src/library/libraryquery.cpp
+++ b/src/library/libraryquery.cpp
@@ -113,7 +113,7 @@ void LibraryQuery::AddWhere(const QString& column, const QVariant& value,
 }
 
 void LibraryQuery::AddCompilationRequirement(bool compilation) {
-  where_clauses_ << QString("effective_compilation = %1").arg(compilation ? 1 : 0);
+  where_clauses_ << QString("effective_compilation LIKE %1").arg(compilation ? 1 : 0);
 }
 
 QSqlQuery LibraryQuery::Exec(QSqlDatabase db, const QString& songs_table,
