--- kfilemetadata-5.42.0/cmake/FindXattr.cmake
+++ kfilemetadata-5.42.0/cmake/FindXattr.cmake
@@ -13,7 +13,7 @@
     set(XATTR_FOUND TRUE)
 else ()
 
-    find_path(XATTR_INCLUDE attr/xattr.h
+    find_path(XATTR_INCLUDE sys/xattr.h
         /usr/include
         /usr/local/include
         ${CMAKE_INCLUDE_PATH}
--- kfilemetadata-5.42.0/src/xattr_p.h
+++ kfilemetadata-5.42.0/src/xattr_p.h
@@ -32,13 +32,11 @@
 #include <sys/types.h>
 #include <sys/xattr.h>
 
-#if defined(Q_OS_ANDROID)
+#if defined(Q_OS_ANDROID) || defined(Q_OS_LINUX)
 // attr/xattr.h is not available in the Android NDK so we are defining ENOATTR ourself
 #ifndef ENOATTR
 # define ENOATTR ENODATA        /* No such attribute */
 #endif
-#else
-#include <attr/xattr.h>
 #endif
 
 #include <errno.h>
